rustls/src/crypto/mod.rs

```diff
@@ -626,6 +604,47 @@ pub fn default_fips_provider() -> CryptoProvider {
     aws_lc_rs::default_provider()
 }
 
+mod static_default {
+    #[cfg(not(feature = "std"))]
+    use alloc::boxed::Box;
+    use alloc::sync::Arc;
+
+    #[cfg(not(feature = "std"))]
+    use once_cell::race::OnceBox;
+    #[cfg(feature = "std")]
+    use once_cell::sync::OnceCell;
+
+    use super::CryptoProvider;
+
+    #[cfg(feature = "std")]
+    pub(crate) fn install_default(
+        default_provider: CryptoProvider,
+    ) -> Result<(), Arc<CryptoProvider>> {
+        PROCESS_DEFAULT_PROVIDER.set(Arc::new(default_provider))
+    }
+
+    #[cfg(not(feature = "std"))]
+    pub(crate) fn install_default(
+        default_provider: CryptoProvider,
+    ) -> Result<(), Arc<CryptoProvider>> {
+        match PROCESS_DEFAULT_PROVIDER.set(Box::new(Arc::new(default_provider))) {
+            Ok(()) => Ok(()),
+            Err(previous_default_provider) => Err(*previous_default_provider),
+        }
```

Comment:
I think this `match` can be replaced with `map_err()`:
```rust
        PROCESS_DEFAULT_PROVIDER.set(Box::new(Arc::new(default_provider))).map_err(|e|*e)
```

Reply:
Done (with a little reformatting), need to rebase (yet) again (yet yet yet again)